store/src/java/com/zimbra/cs/service/mail/ItemAction.java

```diff
@@ -378,6 +394,62 @@ protected ItemActionResult handleTrashOperation(OperationContext octxt, Element
+     *                            owner file is deleted also notify user that file
+     *                            access has been revoked
+     */
+    protected void handleTrashOperationForSharedFile(OperationContext octxt, Mailbox mbox, List<Integer> local,
+            MailItem.Type type, Map<String, Object> context) throws ServiceException, MessagingException {
+        ZimbraSoapContext zsc = getZimbraSoapContext(context);
+        Account authAccount = getAuthenticatedAccount(zsc);
+        Account account = getRequestedAccount(zsc);
+        MailItem[] items = mbox.getItemById(octxt, local, MailItem.Type.UNKNOWN);
+        SendShareNotification sendNotification = new SendShareNotification();
+        for (MailItem item : items) {
+            if (item instanceof Document && item.getACL() != null) {
+                for (Grant grant : item.getACL().getGrants()) {
+                    // form shareinfodata object to send file revoke notification email to each grantee
+                    ShareInfoData sr = new ShareInfoData();
+                    sr.setOwnerAcctId(item.getAccountId());
+                    sr.setOwnerAcctEmail(mbox.getAccount().getMail());
+                    sr.setOwnerAcctDisplayName(mbox.getAccount().getDisplayName() == null ? mbox.getAccount().getMail()
+                            : mbox.getAccount().getDisplayName());
+                    sr.setItemId(item.getId());
+                    sr.setItemUuid(item.getUuid());
+                    sr.setPath(item.getPath());
+                    sr.setFolderDefaultView(Type.DOCUMENT);
+                    sr.setGranteeType(ACL.GRANTEE_USER);
+                    // get grantee email Id to details to which notification email will be sent
+                    Account granteeAccount = Provisioning.getInstance().getAccount(grant.getGranteeId());
+                    sr.setGranteeId(grant.getGranteeId());
+                    sr.setGranteeName(granteeAccount.getName());
+                    MimeMessage mm = null;
+                    mm = sendNotification.generateShareNotification(authAccount, account, sr, null, Action.revoke, null,
+                            null, true);
+                    mbox.getMailSender().sendMimeMessage(octxt, mbox, true, mm, null, null, null, null, false);
+
+                    // create FileSharedWithMeRequest and call FileShareWithMe API to trigger
+                    // revoke/delete action from shareeAccount
+                    FileSharedWithMeRequest req = new FileSharedWithMeRequest();
+                    req.setAction("revoke");
+                    req.setOwnerFileId(item.getId()); // fileId
+                    req.setFileUUID(item.getUuid()); // fileUUid
+                    req.setOwnerAccountId(item.getAccountId()); // remote account Id
+                    Element request = JaxbUtil.jaxbToElement(req);
+                    // proxy request to FileShareWithMe API
```

Comment:
don't see need for comment here, method call itself gives required info about the proxy call for FileShareWithMe API

Reply:
removed the comment from here